### DISCUSSION TITLE
Enable Sticky Keys?

### DISCUSSION THREAD

**Discussion by Qweting** [hyprwm/Hyprland#5405] [Q&A]
Hello. 
I've just installed Hyprland and i was wondering is there a way to enable Sticky keys? I've read in couple of places that you can not enable it and figured it's best if i ask here. 
Edit 1: Accidentally pressed post and didn't have the time to write everything.

**Comment by karnaKali** [comment]
I certainly would like to know this myself as I only have one functional hand to use and it would be impossible to enable some of the key combinations if not all. Hyperland certainly looks to be the dynamic tiling window manager to go for except for the the the lack of accessibility functionality

**Reply by Qweting** [reply]
I couldn't figure out how to get it work on hyprland, but i did find a workaround in case you are interested. 

I use keyd to achieve the sticky keys functionality, i made a post about it on arch forums. 

Here is the github repo: https://github.com/rvaiya/keyd 

I believe I compiled it from source but it is available on pacman as well. If you still need help installing it, i can post my config. 

Good luck!

**Reply by Qweting** [reply]
Here is the config actually, i had on hand

/etc/keyd/default.conf: 
```
[ids]
*

[main]

control = oneshot(control)
meta = oneshot(meta)
shift = oneshot(shift)
rightalt = oneshot(altgr)

```

This will make the buttons behave like "sticky keys", you can add a delay to it (how long it should press the button) but i don't have in this config file. I can update it with my current config tomorrow. But there is info on the git page 🙂

**Reply by karnaKali** [reply]
Works perfectly for me. Now's time to play!

**Reply by Qweting** [reply]
That's awesome! 
Here is my updated config, oneshot_timeout enables the sticky keys behavior for x seconds, in case you were interested!  

    [ids]
    *

    [global]
	#900 millisecond timeout for sticky keys, 
	# works good for keyboard but if you use mouse then 
	# it's a bit problematic. 
    oneshot_timeout = 900ms 


    [main]

    control = oneshot(control)
    meta = oneshot(meta)
    shift = oneshot(shift)
    rightalt = oneshot(altgr)
    leftalt = oneshot(alt)
    capslock = backspace

**Reply by karnaKali** [reply]
Works perfectly! This means everything to me.

**Comment by karnaKali** [comment]
Thanks for the information. It is greatly appreciated.

**Reply by Qweting** [reply]
Absolutely, I'm glad i could help!

**Comment by karnaKali** [comment]
Outstanding! This package gives sooo much power

**Comment by Ryane95** [comment]
Thank you so much this helped a lot. although I am assuming the capslock assignment is optional? I never used Caps Lock anyways

**Reply by Qweting** [reply]
Yeah, I think you might be able to use something else but at the time I didn't know how to do it. If you want to keep Caps Lock then you can  perhaps try this: 

`leftshift = oneshot(shift)`

I haven't tried it myself, It's from their Github page.
